internal/locality/locality.go

```diff
@@ -180,6 +185,29 @@ func ChooseDistributedProcesses(cluster *fdbv1beta2.FoundationDBCluster, process
 		currentLimits[field] = 1
 	}
 
+	if constraint.SelectingCoordinators {
+		// If the cluster runs in a multi-region config and has more than 3 DCs we want to make sure that the primary is
+		// preferred for the 3 coordinators.
+		// See: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/2034
+		if cluster.Spec.DatabaseConfiguration.UsableRegions > 1 && cluster.Spec.DatabaseConfiguration.CountUniqueDataCenters() > 3 {
```

Comment:
This works for some specific setups e.g. 4 different DCs but not for some more exotic setups like having 1 primary dc with 2 different primary satellites and having 1 remote and 2 different remote satellites (that would be 6 unique dcs).